runbotics/runbotics-scheduler/src/scheduler-database/process-credential/process-credential.service.ts

```diff
@@ -77,6 +81,46 @@ export class ProcessCredentialService {
         );
     }
 
+    async update(processId: number, user: User, processCredentialsToUpdate: EditProcessCredentialsDto) {
+        await this.processService.checkAccessAndGetById(processId, user);
+
+        const currentProcessCredentialsForTemplate = await this.findAllByProcessId(
+            processId,
+            user,
+            processCredentialsToUpdate.templateId
+        );
+
+        const intersection = _.intersectionBy(
+            processCredentialsToUpdate.credentials,
+            currentProcessCredentialsForTemplate,
+            'id'
+        );
+
+        if (currentProcessCredentialsForTemplate.length !== intersection.length) {
+            throw new BadRequestException();
+        }
+
+        return this.processCredentialRepository.manager.transaction(
+            async (manager) => {
+                return Promise.all(processCredentialsToUpdate.credentials.map(async (credential) => {
+                    const partial: Partial<ProcessCredential> = {};
+
+                    partial.order = credential.order;
+
+                    return manager
+                        .getRepository(ProcessCredential)
+                        .update({ id: credential.id }, partial);
+                }));
+            }
+        )
+        .then(() => {
+            return this.findAllByProcessId(processId, user);
+        })
+        .catch(() => {
+            throw new BadRequestException(`Failed to update credentials for processId: ${processId}`);
```

Comment:
consider throwing original error or error message If it includes valuable information